testing/unit/resource_test.ts

```diff
@@ -136,6 +136,15 @@ test(async function testDomPoint() {
 
   assert(wrap.undefinedResult() === undefined);
 
+  wrap.withValidateInt(10);
+
+  try {
+    wrap.withValidateInt("bad");
+  } catch (e) {
+    assert(e instanceof TypeError);
+    assertEquals(e.message, "Expected int");
+  }
```

Comment:
I think we need `assertThrows()` helper? If the function doesn't throw at all this test will still pass